File: packages/forms/frontend/sirius-components-forms/src/form/FormContext.ts
/*******************************************************************************
 * Copyright (c) 2023, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import React from 'react';
import { PropertySectionComponentRegistry, PropertySectionContextValue } from './FormContext.types';
import { GQLWidget } from './FormEventFragments.types';

const propertySectionsRegistry: PropertySectionComponentRegistry = {
  getComponent: () => {
    return null;
  },
  getPreviewComponent: (_widget: GQLWidget) => {
    return null;
  },
  getWidgetContributions: () => [],
};

const value: PropertySectionContextValue = {
  propertySectionsRegistry,
};
export const PropertySectionContext = React.createContext<PropertySectionContextValue>(value);

File: packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/FormContext.java
/*******************************************************************************
 * Copyright (c) 2026 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
package org.eclipse.sirius.components.collaborative.forms;

import java.util.List;
import java.util.Objects;

import org.eclipse.sirius.components.forms.Form;
import org.eclipse.sirius.components.forms.description.FormDescription;

/**
 * Used to keep the form representation and its rendering inputs in memory.
 *
 * @author sbegaudeau
 */
public record FormContext(String id, Form form, FormDescription formDescription, Object object, List<Object> selection) {

    public FormContext {
        Objects.requireNonNull(id);
        Objects.requireNonNull(formDescription);
        Objects.requireNonNull(object);
        selection = List.copyOf(selection);
    }

    public FormContext withForm(Form newForm) {
        return new FormContext(this.id, newForm, this.formDescription, this.object, this.selection);
    }
}
